embassy-nrf/src/buffered_uarte.rs

```diff
@@ -429,21 +429,22 @@ impl<'a, U: UarteInstance, T: TimerInstance> Drop for StateInner<'a, U, T> {
     fn drop(&mut self) {
         let r = U::regs();
 
-        // TODO this probably deadlocks. do like Uarte instead.
+        r.inten.reset();
+        r.events_rxto.reset();
+        r.tasks_stoprx.write(|w| unsafe { w.bits(1) });
 
-        self.timer.stop();
```

Comment:
I noticed this part is removed. Shouldn't the timer be stopped as well anymore or is that done implicitly?

Reply:
I'll double check. I figured it was done implicitly since the normal Uarte also doesn't explicitly stop the timer.

Reply:
Turns out I was wrong. The normal Uarte did this with an OnDrop guard. Fixed.